InaraResponder/InaraResponder.cs

```diff
@@ -534,58 +534,64 @@ private void handleShipTransferInitiatedEvent(ShipTransferInitiatedEvent @event)
-            {
-                { "shipType", currentShip.EDName },
-                { "shipGameID", currentShip.LocalId },
-                { "shipName", currentShip.name },
-                { "shipIdent", currentShip.ident },
-                { "shipRole", currentShip.Role.invariantName },
-                { "isHot", currentShip.hot },
+            var currentShipData = new Dictionary<string, object>()
+            {
+                { "shipType", @event.edModel },
+                { "shipGameID", @event.shipid },
+                { "shipName", @event.name },
+                { "shipIdent", @event.ident },
                 { "isCurrentShip", true }
             };
+            var currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
+            if (currentShip?.EDName == @event.edModel)
+            {
+                currentShipData.Add("shipRole", (currentShip?.Role ?? Role.MultiPurpose).invariantName);
+                currentShipData.Add("isHot", currentShip?.hot ?? false);
+            }
             inaraService.EnqueueAPIEvent(new InaraAPIEvent(@event.timestamp, "setCommanderShip", currentShipData));
         }
 
         private void handleShipLoadoutEvent(ShipLoadoutEvent @event)
         {
-            Ship currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
-            Dictionary<string, object> currentShipData = new Dictionary<string, object>()
-            {
-                { "shipType", currentShip.EDName },
-                { "shipGameID", currentShip.LocalId },
-                { "shipName", currentShip.name },
-                { "shipIdent", currentShip.ident },
-                { "shipRole", currentShip.Role.invariantName },
-                { "isHot", currentShip.hot },
+            var currentShipData = new Dictionary<string, object>()
+            {
+                { "shipType", @event.edModel },
+                { "shipGameID", @event.shipid },
+                { "shipName", @event.shipname },
+                { "shipIdent", @event.shipident },
                 { "isCurrentShip", true },
                 { "shipHullValue", @event.hullvalue },
                 { "shipModulesValue", @event.modulesvalue },
                 { "shipRebuyCost", @event.rebuy }
             };
+            var currentShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.shipid);
+            if (currentShip?.EDName == @event.edModel)
+            {
+                currentShipData.Add("shipRole", (currentShip?.Role ?? Role.MultiPurpose).invariantName);
+                currentShipData.Add("isHot", currentShip?.hot ?? false);
+            }
             inaraService.EnqueueAPIEvent(new InaraAPIEvent(@event.timestamp, "setCommanderShip", currentShipData));
 
-            List<Dictionary<string, object>> modulesData = new List<Dictionary<string, object>>();
+            var modulesData = new List<Dictionary<string, object>>();
```

Comment:
What about replacing lines 575-591 with:
```
var modulesData  = @event.hardpoints.Where( h => h != null ).Select( h => GetModuleData(h.name, h.module))
     .Union( @event.compartments. Where( c => c != null ).Select( c => GetModuleData( c.name, c.module));
```
Code alignment in comments is awful... :(

Reply:
We could but that really isn't within scope for what I'm trying to accomplish with this PR and we don't need to adapt our code to use LINQ in every situation.

Reply:
I appreciate this is tangential to the main scope of the PR.
Still, it's an opportunity to improve a bit our codebase. In the long runs, these bits are going to sum up 😉 

Anyway, I respect your call on this.

Reply:
Is it an improvement or a style preference?

Reply:
A bit of both. 😄 

LINQ is highly optimized and can figure out the best pipeline to process the loop.
Also, `List<T>.Add()` can be expensive if the list needs to be reallocated (see the [Remarks](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.add?view=netcore-3.1#remarks) section of `Add`), while LINQ builds its pipeline to then use `AddRange` ([Remarks](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.addrange?view=netcore-3.1#remarks)).
The difference can be big, especially if the list needs to be re-allocated multiple times.
I know multiple reallocations might not be relevant here, but to me, it's simpler to just go for LINQ and do not even worry about differentiating the cases.

And there is also a style preference, probably personal.
To my eye, a LINQ statement is more focused on the goal of the code, as the boilerplate of a `foreach` tends to distract me.
In the specific example, when I am reading through the code, I wonder how many times `modulesData` is used (and I skip ahead), what happens when `hardpoint` is null (and I skip ahead), or `compartment` is null (skip ahead again).
Also, still my personal preference, variables like `hardpoint` and `compartment` really serve no purpose except temporary containers, so I'd rather simplify the code and remove them as much as possible.
Compared with the equivalent var in the LINQ statement, `hardpoint` becomes `h`, and its whole scope is covered with a glance of the eye.

Of course, YMMV, and I am the newest member of the team, so I likely miss many of the established team conventions 😅

Reply:
While I've been with the project for longer, my experience outside of this project is fairly limited. Still plenty for me to learn. Part of my reason for challenging you on this was to better understand your perspective so thank you for taking the time to explain. I'll certainly consider your proposal. :-)

Reply:
Thank YOU for challenging my opinions and for being open and look into new perspectives 😄 

I'll resolve this conversation while you decide what's best for the project.